dbt_platform_helper/utils/validation.py

```diff
@@ -677,13 +679,35 @@ def _validate_environment_pipelines_triggers(config):
         abort_with_error(error_message + "\n  ".join(errors))
 
 
+def lint_yaml_for_duplicate_keys(file_path):
+
+    lint_yaml_config = """
+rules:
+  key-duplicates: enable
+"""
+    yaml_config = config.YamlLintConfig(lint_yaml_config)
+
+    with open(file_path, "r") as yaml_file:
+        file_contents = yaml_file.read()
+        results = linter.run(file_contents, yaml_config)
+
+    parsed_results = [
+        f"Line {result.line}: {result.message} (Severity: {result.level})" for result in results
+    ]
+
+    return parsed_results
+
+
 def load_and_validate_platform_config(
     path=PLATFORM_CONFIG_FILE, disable_aws_validation=False, disable_file_check=False
 ):
     if not disable_file_check:
         config_file_check(path)
     try:
         conf = yaml.safe_load(Path(path).read_text())
+        duplicate_keys = lint_yaml_for_duplicate_keys(path)
+        if duplicate_keys:
+            abort_with_error(f"Error: Duplicate keys found in platform-config: {duplicate_keys}")
```

Comment:
Any way we can retain the info given in parsed results? Very handy for debug

Reply:
Hmm okay I was reading this print statement wrong - it feels like currently if there are many duplicate keys then this error message is the opposite of clear and short as it will print the whole array of parsed results?

Reply:
According to the test below, the error will look like this...

```
Line 98: duplication of key "connors-favourite-key-but-duplicated" in mapping (key-duplicates) (Severity: error)
```

Reply:
The test checks the return of `lint_yaml_for_duplicate_keys` function, not the return or throw of `load_and_validate_platform_config`

Reply:
e.g ran this locally and got this error message: `Error: Error: Duplicate keys found in platform-config: ['Line 83: duplication of key "session" in mapping (key-duplicates) (Severity: error)', 'Line 99: duplication of key "session" in mapping (key-duplicates) (Severity: error)', 'Line 131: duplication of key "session" in mapping (key-duplicates) (Severity: error)', 'Line 147: duplication of key "queue" in mapping (key-duplicates) (Severity: error)']`

Reply:
and was this what you expected? 😢

Reply:
We probably want something like...

```
Error: Duplicate keys found in platform-config:
  Line 83: duplication of key "session"
  Line 99: duplication of key "session"
  Line 131: duplication of key "session"
  Line 147: duplication of key "queue"
```

Reply:
I like that it is verbose but perhaps to keep it clean we could remove the extra `Error` and only print the first item in results?

Even if there are 4 duplicate keys it would only tell you about the first issue each time the code is ran - e.g in this case: `Error: Duplicate keys found in platform-config: Line 83: duplication of key "session" in mapping`

Also not sure if there is any need to display the error type or the severity when we are currently only checking for duplicate keys

Reply:
> We probably want something like...
> 
> ```
> Error: Duplicate keys found in platform-config:
>   Line 83: duplication of key "session"
>   Line 99: duplication of key "session"
>   Line 131: duplication of key "session"
>   Line 147: duplication of key "queue"
> ```

Yeah this is nice to work with + debug